libcontainer/factory_linux.go

```diff
@@ -336,33 +336,30 @@ func (l *LinuxFactory) Type() string {
 // This is a low level implementation detail of the reexec and should not be consumed externally
 func (l *LinuxFactory) StartInitialization() (err error) {
 	var (
-		pipefd, fifofd int
-		consoleSocket  *os.File
-		envInitPipe    = os.Getenv("_LIBCONTAINER_INITPIPE")
-		envFifoFd      = os.Getenv("_LIBCONTAINER_FIFOFD")
-		envConsole     = os.Getenv("_LIBCONTAINER_CONSOLE")
+		envInitPipe = os.Getenv("_LIBCONTAINER_INITPIPE")
+		envInitType = os.Getenv("_LIBCONTAINER_INITTYPE")
+		envFifoFd   = os.Getenv("_LIBCONTAINER_FIFOFD")
+		envConsole  = os.Getenv("_LIBCONTAINER_CONSOLE")
```

Comment:
Now you don't even need `var` block, just `envXXX := os.Getenv(YYY)`.

In fact, I do not like that all four variables are initialized at the start -- I'd rather move them closer to the place of use.
For one thing, `envFifoFd` is only used in case `it == initStandard` so it can be initialized right there under the `if` block.

Reply:
Ah, yes, could be something like

```go
	if c := os.Getenv("_LIBCONTAINER_CONSOLE"); c != "" {
		console, err := strconv.Atoi(c)
		if err != nil {
			return fmt.Errorf("unable to convert _LIBCONTAINER_CONSOLE=%s to int: %s", c, err)
		}
		consoleSocket = os.NewFile(uintptr(console), "console-socket")
		defer consoleSocket.Close()
	}
```

Reply:
ok，I get it, declare variables closer to the place where it use is also perfect.